File: list_1/ProblemC.java
package first_list;

import java.util.Arrays;
import java.util.Scanner;

public class ProblemC {

	public static void main(String[] args) {
		Scanner input = new Scanner(System.in);
		int tests = input.nextInt();
		input.nextLine();
		
		for (int i = 0; i < tests; i++) {
			long[] ns = Arrays.stream(input.nextLine().split(" ")).mapToLong(Long::parseLong).toArray();
			long n = ns[0], steps = ns[1];
			long[] florest = Arrays.stream(input.nextLine().split(" ")).mapToLong(Long::parseLong).toArray();
			long sum = 0, result = 0;
			if (steps < n) {
				for (int j = 0; j < steps; j++) {
					sum += florest[j];
				}
				result = sum;
				for (long j = steps; j < n; j++) {
					sum += florest[(int) j];
					sum -= florest[(int) (j - steps)];
					result = Math.max(result, sum);
				}
				result += (steps*(steps-1))/2; // (n * (n - 1))/2 = a0 + a1 + a2 + a3 + ... + an
				System.out.println(result);
			} else {
				result = (n*(steps-n)) + ((n*(n-1))/2);
				for (int j = 0; j < n; j++) {
					result += florest[j];
				}
				System.out.println(result);
			}
		}
	}
}

File: list_1/ProblemC.py
tests = int(input())

for i in range(tests):
    n, steps = map(int, input().split())
    florest = [int(n) for n in input().split()]
    if (steps <= n):
        sum_ = 0
        for j in range(steps):
            sum_ += florest[j]
        result = sum_
        for j in range(steps, n, 1):
            sum_ += florest[j]
            sum_ -= florest[j - steps]
            result = max(result, sum_)
        
        result += (steps*(steps-1))/2
        print(int(result))
    else:
        result = (n*(steps-n)) + ((n*(n-1))/2)
        for j in range(n):
            result += florest[j]
        print(int(result))
